device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/ModuleClient.java

```diff
@@ -146,33 +147,54 @@ private ModuleClient(String hostName, String deviceId, String moduleId, SasToken
 
     /**
      * Create a module client instance from your environment variables
+     * @param unixDomainSocketChannel the implementation of the {@link UnixDomainSocketChannel} interface that will be used if any
+     * unix domain socket communication is required. May be null if no unix domain socket communication is required. If
+     * this argument is null and unix domain socket communication is required, this method will through an {@link IllegalArgumentException}.
+     * To check if unix domain socket communication is required for your Edge runtime, check its "IOTEDGE_WORKLOADURI"
+     * environment variable. If it is not present, or its value is prefixed with "HTTP" or "HTTPS", then no unix domain
+     * socket communication is required, and this argument can be set to null. If its value is present and is prefixed
+     * with "unix", then unix domain socket communication will be required, and this argument must not be null.
```

Comment:
Nice explanation!
Q - if we can create a module client without needing to specify the unix socket domain channel, do we still have a no-args constructor? It'll save users from having to pass in `null`.

Reply:
We can add one later if asked for it, but I'd rather users be aware that the unix domain socket communication may be necessary. I imagine users would gravitate towards the simplest constructor as their first step, and get confused when they get an error message saying "must provide a unix domain socket implementation"